### DISCUSSION TITLE
How to figure out if running on http vs https?

### DISCUSSION THREAD

**Discussion by marcofranssen** [vercel/next.js#21488] [Help] [answered]
In my api page I'm doing the following:

```js
export default function handler(req, res) {
  const host = req.headers.host;
  const proto = req.connection.encrypted ? "https" : "http";

  //construct url for xml sitemaps
}
```

However when deploying this code on vercel it seems the request is never encrypted. What would be a Vercel compatible way of checking the protocol?

**Comment by suyundukov** [comment, marked as answer]
Hey, @marcofranssen. You can check `X-Forwarded-Proto` header. As Vercel runs your app behind proxy server like Nginx, it's the only way to get request protocol:

```js
const proto = req.headers['x-forwarded-proto'];
```

**Reply by marcofranssen** [reply]
I think my current code also isn't working because of the proxy. How do I enable this `trust proxy`? What do you mean by that?

**Reply by suyundukov** [reply]
Was wrong about `req.protocol` and `req.secure`. I edited my comment. Sorry for misleading you.

**Comment by cristian-eriomenco** [comment]
What the fuck do you want from me Nextjs:
![image](https://github.com/user-attachments/assets/83481d82-f302-4075-bc9a-767aa2182581)
